src/Features/CSharp/Portable/ConvertToRecord/ConvertToRecordCommon.cs

```diff
@@ -394,6 +426,81 @@ private static SyntaxList<AttributeListSyntax> GetModifiedAttributeListsForPrope
+            var projectLookup = referenceLocations.ToLookup(refLoc => refLoc.Document.Project.Id);
+            foreach (var (projectID, projectLocations) in projectLookup)
+            {
+                // organize by project first, so we can solve one project at a time
+                var project = solutionEditor.OriginalSolution.GetRequiredProject(projectID);
+                if (project.Language != LanguageNames.CSharp)
+                {
+                    // since this is a CSharp-dependent file, we can't fix VB projects that 
+                    continue;
+                }
+
+                var compilation = await project
+                    .GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
+                var docLookup = projectLocations.ToLookup(refLoc => refLoc.Document.Id);
+                foreach (var (documentID, documentLocations) in docLookup)
+                {
+                    var documentEditor = await solutionEditor
+                        .GetDocumentEditorAsync(documentID, cancellationToken).ConfigureAwait(false);
+                    var syntaxTree = await documentEditor.OriginalDocument
+                        .GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+                    var semanticModel = compilation.GetSemanticModel(syntaxTree);
+                    var objectCreationExpressions = documentLocations
+                        // we should find the identifier node of an object creation expression
+                        .Select(referenceLocations => referenceLocations.Location.FindNode(cancellationToken).Parent)
+                        .OfType<ObjectCreationExpressionSyntax>();
+                    foreach (var objectCreationExpression in objectCreationExpressions)
+                    {
+                        var (constructorArgs, nodesToRemove) =
+                            ConvertToRecordHelpers.GetConstructorArgumentsFromObjectCreation(
+                                objectCreationExpression, positionalParameters, semanticModel, cancellationToken);
+
+                        if (!constructorArgs.IsDefaultOrEmpty)
+                        {
+                            var initializerExpressions = objectCreationExpression.Initializer!.Expressions;
+                            // need to get all indices and then remove because otherwise nodes change
+                            var removalIndices = nodesToRemove
+                                .SelectAsArray(initializerExpressions.IndexOf)
+                                .OrderByDescending(i => i);
+                            // remove in descending order so indices aren't shifted around
+                            foreach (var nodeIndex in removalIndices)
+                            {
+                                initializerExpressions = initializerExpressions.RemoveAt(nodeIndex);
+                            }
```

Comment:
This didn't work since removing a node recreated the entire syntax list so the nodes in our original list were no longer the same as the ones in initializerExpressions. However, I can call `RemoveNodes` on the Initializer syntax (parent of the expressions list), then handle the empty case afterwards.